### DISCUSSION TITLE
How to use same keybindings from terminal to toggle the terminal

### DISCUSSION THREAD

**Discussion by luisiacc** [folke/snacks.nvim#62] [Q&A]
I have this config
```lua
{
    "folke/snacks.nvim",
    priority = 1000,
    lazy = false,
    opts = {
      -- your configuration comes here
      -- or leave it empty to use the default settings
      -- refer to the configuration section below
    },
    keys = {
      {
        "<C-y>",
        function()
          Snacks.terminal.toggle()
        end,
        desc = "Terminal",
      },
    },
  }
```

To toggle the terminal I use C-y, I want to press C-y to also close the terminal when I'm inside it, but this does not seem to work, I have to press double <esc> in order to be able to press C-y to toggle it.

**Comment by folke** [comment]
add `mode = {"n", "t"}`, like in the example

**Reply by xian-wen** [reply]
Sorry, this works when toggling terminal in cwd, but not for root dir. If I press the keymap in terminal mode again, it just opens a second terminal in cwd; after that, if I press the keymap again, the second cwd terminal can be toggled. The first root-dir terminal can never be toggled.
![image](https://github.com/user-attachments/assets/2022b570-a55f-4d4a-8455-e365ef8a5a5f)

**Reply by xian-wen** [reply]
But the default way in LazyVim, i.e., mapping `<C-/>` to `close` in terminal mode, does work.
